src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,250 @@
+            String[] contentBy = typeContentBy[1].trim().split("/[Bb][Yy]", 2);
+            if (contentBy[0].trim().equals("") || contentBy[1].trim().equals("")) {
+                throw new EmptyDescriptionException(CommandType.DEADLINE);
+            }
+            Deadline d = tasks.addDeadline(contentBy[0].trim(), contentBy[1].trim());
+            showAddResult(d);
+            tasks.writeToTxt(filePath);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            showExecuteResult("OOPS!!! No /by founded in the command");
+
+        }
+
+    }
+
+    public static void executeAddEvent(String userCommand) throws EmptyDescriptionException, IOException {
+        try {
+            String[] typeContentAt= userCommand.trim().split("[Ee][Vv][Ee][Nn][Tt]", 2);
+            String[] contentAt = typeContentAt[1].trim().split("/[Aa][Tt]", 2);
+            if (contentAt[0].trim().equals("") || contentAt[1].trim().equals("")) {
+                throw new EmptyDescriptionException(CommandType.EVENT);
+            }
+            Event e = tasks.addEvent(contentAt[0].trim(), contentAt[1].trim());
+            showAddResult(e);
+            tasks.writeToTxt(filePath);
+        } catch (IndexOutOfBoundsException e){
+            showExecuteResult("OOPS!!! No /at founded in the command");
+        }
+
+    }
+
+    public static void executeList(String userCommand) {
+        tasks.listAllTasks();
+    }
+
+    public static void executeDone(String userCommand) throws EmptyDescriptionException, IOException {
+        try{
+            String[] typeIndex = userCommand.split("[Dd][Oo][Nn][Ee]",2);
+            int taskIndexShow = Integer.parseInt(typeIndex[1].trim());
+            if(taskIndexShow <= 0 || taskIndexShow > tasks.getNumOfTasks()) {
+                throw new EmptyDescriptionException(CommandType.DONE);
+            }
+            tasks.markTaskDone(taskIndexShow);
+            tasks.writeToTxt(filePath);
+        } catch (NumberFormatException e) {
+            throw new EmptyDescriptionException(CommandType.DONE);
+        }
+    }
+
+    public static void executeDELETE(String userCommand) throws EmptyDescriptionException, IOException {
+        try{
+            String[] typeIndex = userCommand.split("[Dd][Ee][Ll][Ee][Tt][Ee]",2);
+            int taskIndexShow = Integer.parseInt(typeIndex[1].trim());
+            if(taskIndexShow <= 0 || taskIndexShow > tasks.getNumOfTasks()) {
+                throw new EmptyDescriptionException(CommandType.DELETE);
+            }
+            tasks.deleteTask(taskIndexShow);
+            tasks.writeToTxt(filePath);
+        } catch (NumberFormatException e) {
+            throw new EmptyDescriptionException(CommandType.DELETE);
+        }
+    }
+
+    private static void executeExitProgramRequest() {
+        showBye();
+        System.exit(0);
+    }
+
+    public static void executeCommand(String userInputString) throws IOException {
```

Comment:
Try to avoid long methods with more than 30 lines of code. Could make use of more abstraction